### PR TITLE
Version bump and README updates for Bandits.

### PR DESCRIPTION
**Pull request by typotter**

### CODE REVIEW

File: dot-net-sdk/EppoClientConfig.cs [hunk 1]
```diff
@@ -1,11 +1,32 @@
...
-        IAssignmentLogger AssignmentLogger)
+    public string ApiKey;
+    public readonly IAssignmentLogger AssignmentLogger;
+
+    public EppoClientConfig(string apiKey,
+                            IAssignmentLogger? assignmentLogger)
     {
-        public string BaseUrl { get; set; } = Constants.DEFAULT_BASE_URL;
+        this.ApiKey = apiKey;
+        this.AssignmentLogger = assignmentLogger ?? new DefaultLogger();
```
Comment (typotter) [inline dot-net-sdk/EppoClientConfig.cs:16]: Refactored here to allow assignmentLogger param to be nullable and set a noop logger in its place.

File: dot-net-sdk/dto/AssignmentLogData.cs [hunk 2]
```diff
@@ -1,8 +1,9 @@
 using System.Reflection.Metadata.Ecma335;
+using System.Runtime.Serialization;
 
 namespace eppo_sdk.dto;
 
-public record AssignmentLogData
+public record AssignmentLogData : ISerializable
```
Comment (typotter) [inline dot-net-sdk/dto/AssignmentLogData.cs:6]: ISerializable lets the object be passed directly to Segment analytics SDK, and possibly others.

File: dot-net-sdk/dto/bandit/BanditLogData.cs [hunk 3]
```diff
@@ -8,11 +10,11 @@ public record BanditLogEvent(string FlagKay,
                              double? OptimalityGap,
                              string? ModelVersion,
                              DateTime Timestamp,
-                             IReadOnlyDictionary<string, double>? subjectNumericAttributes,
```
Comment (typotter) [inline dot-net-sdk/dto/bandit/BanditLogData.cs:11]: sneakyCasing => FixedCasing

File: dot-net-sdk/eppo-sdk.csproj [hunk 4]
```diff
@@ -9,7 +9,7 @@
     <PackageProjectUrl>https://github.com/Eppo-exp/dot-net-server-sdk</PackageProjectUrl>
     <RepositoryUrl>https://github.com/Eppo-exp/dot-net-server-sdk</RepositoryUrl>
     <PackageId>Eppo.Sdk</PackageId>
-    <Version>2.1.0</Version>
+    <Version>3.0.0</Version>
```
Comment (typotter) [inline dot-net-sdk/eppo-sdk.csproj:12]: or do we jump to v4?

File: dot-net-sdk/dto/AssignmentLogData.cs (hunk 2 above)
Comment (aarsilv) [reply dot-net-sdk/dto/AssignmentLogData.cs:6]: Great thinking!

File: dot-net-sdk/dto/AssignmentLogData.cs [hunk 5]
```diff
@@ -35,4 +36,16 @@ public record AssignmentLogData
         MetaData = metaData;
         ExtraLogging = extraLoggging;
     }
+    public void GetObjectData(SerializationInfo info, StreamingContext context)
```
Comment (aarsilv) [inline dot-net-sdk/dto/AssignmentLogData.cs:39]: super minor but should there a blank line between this and the previous method?

File: dot-net-sdk/eppo-sdk.csproj (hunk 4 above)
Comment (aarsilv) [reply dot-net-sdk/eppo-sdk.csproj:12]: 3 is fine!

**Review by aarsilv — APPROVED** [review]

File: dot-net-sdk/dto/AssignmentLogData.cs (hunk 5 above)
Comment (typotter) [reply dot-net-sdk/dto/AssignmentLogData.cs:40]: 📇

**Pull request merged**